### ISSUE TITLE
Several segfault gathered when playing multiplayer

### ISSUE THREAD

**Issue by Bertram25** [OpenDungeons/OpenDungeons#47]
Hey,

We've noted down a few segfault causes while playing:
- [x] Typing too fast in the chat makes the key handled as normal keys. Thus if you type QS too quicky the camera will become crazy. #49 
- [x] Using too long nicknames or nicknames with spaces makes the client segfault. (Seems ok, please reopen a new issue if it still happens.)
- [x] Typing special characters in chat makes the client segfault. #48 
- [x] Adding a quarter room makes the server crash. Might be a bug in the room code, though. #44 
- [x] Such segfault could be seen at random, but not only that one: (Closing as duplicated of #69)
  [bt]: (1) /usr/lib/x86_64-linux-gnu/libstdc++.so.6 : std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&)+0x8 [0x7fc9cc5e80d8]
  [bt]: (2) /usr/lib/x86_64-linux-gnu/libstdc++.so.6 : std::basic_string<char, std::char_traits<char>, std::allocator<char> >::basic_string(std::string const&)+0x8 [0x7fc9cc5e80d8]
  [bt]: (3) ./OpenDungeons.bin : ODServer::notifyClientMessage(ODSocketClient*)+0x68 [0x5d9a88]
  [bt]: (4) ./OpenDungeons.bin : ODSocketServer::doTask(int)+0xfc [0x5dc95c]
  [bt]: (5) ./OpenDungeons.bin : ODServer::serverThread()+0x42 [0x5db712]
  [bt]: (6) /usr/lib/x86_64-linux-gnu/libsfml-system.so.2 : +0x6508 [0x7fc9cca54508]
  [bt]: (7) /lib/x86_64-linux-gnu/libpthread.so.0 : +0x80a4 [0x7fc9cce720a4]
  [bt]: (8) /lib/x86_64-linux-gnu/libc.so.6 : clone+0x6d [0x7fc9cbd4e04d]

It seems I missed debug symblos for some obscure reasons, sorry.

**Comment by akien-mga** [comment]
See issue #49 for the first point. Note that it AFAIK did not trigger segfaults.

See issue #48 for the third point.

**Comment by akien-mga** [comment]
Backtrace of a segfault that I got in a single player game after 10 minutes or so:

```
signal 11 (Segmentation fault), address is 0x80 from 0x5ca3ca

[bt]: (1) ./OpenDungeons.bin : MovableGameEntity::update(float)+0x2a [0x5ca3ca]
[bt]: (2) ./OpenDungeons.bin : MovableGameEntity::update(float)+0x2a [0x5ca3ca]
[bt]: (3) ./OpenDungeons.bin : GameMap::updateAnimations(float)+0x6e [0x58fc7e]
[bt]: (4) ./OpenDungeons.bin : ODFrameListener::frameStarted(Ogre::FrameEvent const&)+0x1fc [0x5e2d9c]
[bt]: (5) /lib64/libOgreMain.so.1.9.0 : Ogre::Root::_fireFrameStarted(Ogre::FrameEvent&)+0x44 [0x7fbef8033f04]
[bt]: (6) /lib64/libOgreMain.so.1.9.0 : Ogre::Root::_fireFrameStarted()+0x21 [0x7fbef80357e1]
[bt]: (7) /lib64/libOgreMain.so.1.9.0 : Ogre::Root::renderOneFrame()+0x9 [0x7fbef80358d9]
[bt]: (8) /lib64/libOgreMain.so.1.9.0 : Ogre::Root::startRendering()+0x2d [0x7fbef803592d]
[bt]: (9) ./OpenDungeons.bin : ODApplication::ODApplication()+0x3f8 [0x5cd508]
[bt]: (10) ./OpenDungeons.bin : main+0x1e [0x467ebe]
[bt]: (11) /lib64/libc.so.6 : __libc_start_main+0xf0 [0x7fbef5208f90]
[bt]: (12) ./OpenDungeons.bin() [0x4692b4]
```

I might be missing some debug symbols too, btw.

**Comment by Bertram25** [comment]
Ok, let this issue be only about segfaults. Sorry, I mixed things here and it's a messy practice.

**Comment by akien-mga** [comment]
I've just played for half an hour on local multiplayer (hosting the server and the client on the same machine), and I experienced no segfault :-)
I can't tell what was different this time, maybe something in the AI behaviour?

**Comment by hwoarangmy** [comment]
1 - Fixed
2 - Cannot reproduce. But I have added checks to make sure no special chars are used for nicknames
3 - Fixed
4 - Cannot reproduce even when there is a room absorbtion
5 - Don't know what's going on
6 - Don't know what's going on

Next time, please open different issues for different problems so that we can assign and close them when they are fixed.

**Comment by hwoarangmy** [comment]
Btw, I've already let the game run with local network during something like 1-2 hours with no crash. But there may be differences when we use global network...

**Comment by akien-mga** [comment]
Note that I created issue #48 for (3) and #49 for (1), you can use those for reference.

**Comment by hwoarangmy** [comment]
Yes, I have already assigned them to me :)
But by splitting issues, we can close them more easily and it avoids to have too much open issues because one point over five is not closed

**Comment by akien-mga** [comment]
Agreed. I'm sure Bertram was just really eager to report the few issues we found during our nice playtest :-)

**Comment by Bertram25** [comment]
> Next time, please open different issues for different problems so that we can assign and close them when they are fixed.

As I said above, 

> Sorry, I mixed things here and it's a messy practice.

Again, I'm sorry for mixing things up. IIRC, I had to go to bed, and wanted the stuff to be written down somewhere. I'm not kind doing this usually, anyway. ;)

**Comment by Bertram25** [comment]
If you prefer it, we can close this one and only open new ones about the remaining open points. Somehow, this is what I usually when an issue becomes messy or is too big.

**Comment by hwoarangmy** [comment]
No, don't take me wrong. It's just for next time, no need to do anything. Eventually, we can split the segfault as I guess we have no clue and they might stay open for a while.
I was just thinking that it would be nice to use github to assign tasks so that everybody knows who is doing what and avoiding doing same stuff

**Comment by hwoarangmy** [comment]
BTW, Bertram, are you working on animating training room ? If not, may I have a look ?

**Comment by Bertram25** [comment]
> I was just thinking that it would be nice to use github to assign tasks so that everybody knows who is doing what and avoiding doing same stuff

Part of the oh-so-evil plan. ;) I've seen you've assigned yourself tasks you're on, and I second that practice.

> BTW, Bertram, are you working on animating training room ? If not, may I have a look ?

About OD, I'm officially on four tasks:
- Make a new in game window skin and a few controls. #5 
- Add an entities.cfg file where entities parameters are parsed, such as the model file, the normal texture, the scaling, the default rotation maybe, ... and adapt the rest of the code. So artists will be able to change whatever they want without rebuilding. #60
- Add the hatchery models and button, even if for now, the room won't do much. #61
- Add a tutorial about how the buttons where made in the wiki and add the missing buttons (for traps, but also workers, and the spawn worker spell needed for later...) So one can start working on a button permitting to drag workers from the button, and on the worker spawning spell code. #8 

So feel free to have a look at triggering the animations. :)

I'll add the corresponding issues if they're missing and assign them to me. :)
I'll try to finish that before taking something else, except for urgencies, such as debugging the network code or such.

Also I wanted to say that I'm sorry not to be more active on OD, but atm, @akien-mageia has popped-up a lot of new issues on Valyria Tear (and unfortunately each and every ones are sensible) so I'll need to deal with this in a higher priority to finally get the Episode I out. Then, hopefully, I'll have a bit more time for OD.

regards,

**Comment by hwoarangmy** [comment]
OD is a free time project so if you have something else to do, nobody should say anything. Moreover, you are active enough so that you don't have to apologize when you cannot work on the project.

**Comment by Bertram25** [comment]
Thanks :)

**Comment by hwoarangmy** [comment]
I think I've understood the segfault Akien had. It happens when room objects are deleted (There was 1 place where I forgot to remove them from the gamemap). Patch is on its way :)

**Comment by Bertram25** [comment]
Oh cool! I'm eager to see that! :)
Off-topic: I think we're close to a RC2 for VT. :D

**Comment by Bertram25** [comment]
@hwoarangmy  @akien-mageia 
I won't be available this Week-End. See you on monday! :)

**Comment by akien-mga** [comment]
I can't seem to reproduce the issue in point 2 (Using too long nicknames or nicknames with spaces makes the client segfault.). Actually I was able to reproduce a segfault upon starting a local multiplayer (hosting server + client in two game instances) with funny names, but further attempts at determining what causes a segfault exactly were fruitless.

I tried to set names with tons of accentuated characters and funny unicode symbols, the game starts properly (the "illegal" characters in the name are replaced by spaces in the chatbox).

**Comment by Bertram25** [comment]
> I tried to set names with tons of accentuated characters and funny unicode symbols, the game starts properly (the "illegal" characters in the name are replaced by spaces in the chatbox).

That's a good thing. :)

**Comment by Bertram25** [comment]
> Actually I was able to reproduce a segfault upon starting a local multiplayer (hosting server + client in two game instances) with funny names, but further attempts at determining what causes a segfault exactly were fruitless.

Is this still reproducible? Could you open a new issue with a step by step description?
Thanks! :)

**Comment by Bertram25** [comment]
Every issues in this one seem to be fixed or opened as other issues. I'll close it.

**Comment by akien-mga** [comment]
It's not reproducible no, what I meant was that I experienced a crash once (IIRC at my first test, typing something competely random). Afterwards I did 20+ test with all sorts of cleverly-crafted names to indentify faulty strings, but I could not reproduce a crash. Further attempts with random names did not trigger a crash either.

So it's good to be closed yes :-)
As for the unknown segfaults in point 5, I guess they're fixed too. We'll see that in our next multiplayer game :-)
